podcasts/End of Year/EndOfYear.swift

```diff
@@ -184,3 +184,59 @@ private class FakeViewController: UIViewController {
         onDismiss?()
     }
 }
+
+extension EndOfYear {
+    static let eoyEligibilityDidChange = NSNotification.Name(rawValue: "eoyEligibilityDidChange")
+
+    private class EligibilityChecker {
+        deinit {
+            notifications.forEach { notificationCenter.removeObserver($0) }
+        }
+
+        var isEligible = false
+
+        private let updateQueue = OperationQueue()
+        private let notificationCenter: NotificationCenter
+        private var notifications: [NSObjectProtocol] = []
+
+        init?(notificationCenter: NotificationCenter = .default) {
+            guard FeatureFlag.endOfYear.enabled else { return nil }
+
+            self.notificationCenter = notificationCenter
+
+            startListening()
+            update()
+        }
+
+        private func startListening() {
+            // The notifications to update the state for
+            let notifications: [Notification.Name] = [
+                // Check after a sync succeeds or the user logs into an account
+                ServerNotifications.syncCompleted,
+                .userSignedIn,
+
+                // Check as the user is listening to episodes
+                Constants.Notifications.playbackPaused,
+                Constants.Notifications.playbackEnded,
+                Constants.Notifications.playbackTrackChanged
+            ]
+
+            self.notifications = notifications.map {
+                notificationCenter.addObserver(forName: $0, object: nil, queue: updateQueue) { [weak self] notification in
```

Comment:
The app is freezing for me due to the use of `OperationQueue`. I wasn't able to found out why though. :(

Here are the steps to reproduce:

1. Clean install
2. Play any episode
3. Move the scrubber
4. Pause
5. 🥶 

Here's a video showing that:

https://github.com/Automattic/pocket-casts-ios/assets/7040243/184724e8-2e45-4de7-a8c1-b78524a06148

If I change `updateQueue` to `.main` it works just fine.